### PR TITLE
[Office-js] Adding support details to shared API enums

### PR DESCRIPTION
**Pull request by AlexJerabek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -279,9 +279,25 @@ declare namespace Office {
...
      * @remarks
-     * <table><tr><td>Hosts</td><td>Access, Excel, Outlook, PowerPoint, Project, Word</td></tr></table>
      * 
      * Returned by the status property of the AsyncResult object.
+     * 
+     * **Support details**
+     * 
+     * A capital Y in the following matrix indicates that this enumeration is supported in the corresponding Office host application. An empty cell indicates that the Office host application doesn't support this enumeration.
+     * 
+     * For more information about Office host application and server requirements, see [Requirements for running Office Add-ins](https://docs.microsoft.com/office/dev/add-ins/concepts/requirements-for-running-office-add-ins).
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts]: need to update the link notation, right?

Comment (AlexJerabek) [reply types/office-js/index.d.ts]: Oh you're totally right. Good catch.

**Comment by AlexJerabek** [comment]
Please refrain from checking in until @ElizabethSamuel-MSFT approves.

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Comment by AlexJerabek** [comment]
Good to merge!

**Pull request merged**